lib/Backend/LinearScan.cpp

```diff
@@ -5135,8 +5135,10 @@ void LinearScan::GeneratorBailIn::BuildBailInSymbolList(
 
         if (unrestorableSymbols.TestAndClear(value->m_id))
         {
-            if (this->NeedsReloadingSymWhenBailingIn(copyPropSym.Key()))
+            if (this->NeedsReloadingSymWhenBailingIn(copyPropSym.Value()))
```

Comment:
Compare with lines 5148-5151.

Important point is we must check if this Sym needs reloading based on its BackEndId.